### PR TITLE
Feat(eslint-plugin-fiori-tools): add node location in the json file

### PR DESCRIPTION
**Pull request by AlinaGovoruhina**
### New Feature

✨ Enhanced the `eslint-plugin-fiori-tools` to include precise source location (`loc`) data in all manifest-related ESLint diagnostics. This enables ESLint to accurately highlight the exact position in a JSON file where a configuration issue occurs, improving developer experience by pinpointing problematic nodes directly in the editor.

The key addition is a `getNode` method on `FioriJSONSourceCode` that traverses the JSON AST to retrieve the node at a given configuration path. If the exact node is not found, it falls back to the closest parent node.

### Changes

* `source-code.ts`: Added `getNodeOrParent` (private) and `getNode` (public) methods to traverse the JSON AST (`Object` and `Member` nodes) by path. Also imports `MemberNode` and `ValueNode` from `@humanwhocodes/momoa`.

* `diagnostics.ts`: Changed `loc` in `ManifestPropertyDiagnosticData` from optional (`loc?: SourceLocation`) to required (`loc: SourceLocation`).

* `helpers.ts`: Updated `checkAppTablesConfiguration` to accept and forward a `FioriJSONSourceCode` instance to the `checkConfiguration` callback.

* `sap-anchor-bar-visible.ts`: Added `FioriJSONSourceCode` guard, resolves `anchorBarVisible` node location via `getNode`, and attaches `loc` to the diagnostic. Refactored page filtering using `.filter()`.

* `sap-condensed-table-layout.ts`: Added `FioriJSONSourceCode` guard and propagated `sourceCode` through `checkCondensedTableLayoutForApp`, `checkCondensedTableLayoutV4`, and `checkCondensedTableLayoutV2`. Each diagnostic now includes `loc`.

* `sap-copy-to-clipboard.ts`: Added `FioriJSONSourceCode` guard and `sourceCode` parameter to `checkConfiguration`, resolving the copy/paste node location and attaching `loc`.

* `sap-creation-mode-for-table.ts`: Propagated `sourceCode` through `reportDiagnostic`, `checkAnalyticalTableV2/V4`, `validateCreateModeV2/V4`, and `processTableV2/V4`. Each diagnostic now resolves and attaches `loc`. Refactored to use destructured `createMode` object for V2 analytical table checks.

* `sap-enable-export.ts` / `sap-enable-paste.ts`: Added `FioriJSONSourceCode` guard and `sourceCode` parameter to `checkConfiguration`; node location is resolved and attached to diagnostics.

* `sap-flex-enabled.ts`: Added `FioriJSONSourceCode` guard; resolves and attaches `loc` of the `flexEnabled` node.

* `sap-state-preservation-mode.ts`: Added `FioriJSONSourceCode` guard; resolves node and passes it to `checkInvalidMode` and `checkModeRecommendation` for `loc` attachment.

* `sap-strict-uom-filtering.ts`: Added `FioriJSONSourceCode` guard; resolves and attaches `loc` for `disableStrictUomFiltering`.

* `sap-table-column-vertical-alignment.ts`: Added `FioriJSONSourceCode` guard; resolves and attaches `loc` for the `tableColumnVerticalAlignment` config path.

* `sap-table-personalization.ts`: Added `FioriJSONSourceCode` guard; removed the previous approach of setting `loc` in the visitor handler; now resolves node locations during `check` phase via `getNode`.

* `sap-width-including-column-header.ts`: Propagated `sourceCode` through `checkTable` and `checkTablesInPage`; resolves `widthIncludingColumnHeader` node location and attaches `loc`.

* **Test files** (`sap-anchor-bar-visible.test.ts`, `sap-flex-enabled.test.ts`, `sap-table-column-vertical-alignment.test.ts`): Added test cases for non-JSON source code, V2/V4 manifest types, and other edge cases to improve coverage.

- [ ] 🔄 Regenerate and Update Summary

<details>
<summary>PR Bot Information</summary>

**Version:** `1.22.0`

- File Content Strategy: Full file content
- LLM: `anthropic--claude-4.6-sonnet`
- Output Template: [Default Template](https://github.tools.sap/intelligent-insights/i2-pull-request/blob/main/src/services/llm/prompts/summary_default_output_template.md)
- Event Trigger: `pull_request.edited`
- Correlation ID: `bf27fb22-903f-4a1e-b214-ec04923ffffe`
- Summary Prompt: [Default Prompt](https://github.tools.sap/intelligent-insights/i2-pull-request/blob/main/src/services/llm/prompts/summary_instructions_prompt.md)
</details>

### CODE REVIEW

File: packages/eslint-plugin-fiori-tools/test/language/json/source-code.test.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            text: jsonText,
+            ast,
+            projectContext: DUMMY_PROJECT_CONTEXT,
+            uri: ''
+        });
+
+        expect(sourceCode.getNode(ast.body, ['path', 'to', 'node'])).toBe(ast.body);
+    });
+
+    it('should get found node', () => {
```
Comment (marcovieth) [inline packages/eslint-plugin-fiori-tools/test/language/json/source-code.test.ts]: It seems that this is also a fallback/not-found test.
Do we also need tests for multi/single segment path match and partial match?

**Review by marcovieth — COMMENTED** [review]
So we fix a bug with a feature.
I have a comment on the tests.

File: packages/eslint-plugin-fiori-tools/test/language/json/source-code.test.ts (hunk 1 above)
Comment (AlinaGovoruhina) [reply packages/eslint-plugin-fiori-tools/test/language/json/source-code.test.ts]: I reworked the tests and covered some new cases.

**Review by marcovieth — DISMISSED** [review]
Code looks good to me.
Tests with good coverage of several cases are there.
Didn't run the tests.

**Review by marcovieth — APPROVED** [review]
Still looks good to me.

**Review by igorsVoronins — APPROVED** [review]
Tested the PR with multiple app types (V2, V4, and CAP). ESLint warnings are now displayed correctly

Image
